custom_components/bmw_connected_drive/sensor.py

```diff
@@ -66,21 +161,55 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
                     account, vehicle, attribute_name, attribute_info
                 )
                 devices.append(device)
+        # LastTrip
+        if vehicle.has_statistics_service:
+            for attribute_name in ATTR_LAST_TRIP:
+                device = BMWConnectedDriveSensor(
+                    account, vehicle, attribute_name, attribute_info, LAST_TRIP
+                    )
+                devices.append(device)
+        # AllTrips
+        if vehicle.has_statistics_service:
+            for attribute_name in ATTR_ALL_TRIPS:
+                device = BMWConnectedDriveSensor(
+                    account, vehicle, attribute_name, attribute_info, ALL_TRIPS
+                    )
+                devices.append(device)
+        # ChargingProfile
+        if vehicle.has_weekly_planner_service:
+            for attribute_name in ATTR_CHARGING_PROFILE:
+                device = BMWConnectedDriveSensor(
+                    account, vehicle, attribute_name, attribute_info, CHARGING_PROFILE
+                    )
+                devices.append(device)
+        # Destinations
+        if vehicle.has_destination_service:
+            for attribute_name in ATTR_DESTINATIONS:
+                device = BMWConnectedDriveSensor(
+                    account, vehicle, attribute_name, attribute_info, DESTINATIONS
+                )
+                devices.append(device)
+
     async_add_entities(devices, True)
 
 
 class BMWConnectedDriveSensor(Entity):
     """Representation of a BMW vehicle sensor."""
 
-    def __init__(self, account, vehicle, attribute: str, attribute_info):
+    def __init__(self, account, vehicle, attribute: str, attribute_info, service=None):
         """Initialize BMW vehicle sensor."""
         self._vehicle = vehicle
         self._account = account
         self._attribute = attribute
+        self._service = service
         self._state = None
-        self._name = f"{self._vehicle.name} {self._attribute}"
-        self._unique_id = f"{self._vehicle.vin}-{self._attribute}"
         self._attribute_info = attribute_info
+        if self._service:
+            self._name = f"{self._vehicle.name} {self._service.lower()}_{self._attribute}"
+            self._unique_id = f"{self._vehicle.vin}-{self._service.lower()}-{self._attribute}"
```

Comment:
Is adding the service needed? We need unique names here, but that is already the case with the attribute isn't it?

Reply:
No, this is needed because some attributes have the same name, like "average_combined_consumption", "average_electric_consumption" &  "average_recuperation"; they exist in both LastTrip & AllTrips...